pennylane/qnn/keras.py

```diff
@@ -347,7 +396,8 @@ def call(self, inputs):
 
         # reshape to the correct number of batch dims
         if has_batch_dim:
-            results = tf.reshape(results, (*batch_dims, *results.shape[1:]))
+            new_shape = tf.concat([batch_dims, tf.shape(results)[1:]], axis=0)
```

Comment:
Why is this change necessary? Is saving impossible without?

Reply:
It's necessary because the shapes need to be Tensorflow tensors in order for `None` dimensions to be compiled properly. Using tuples for the shapes does not work.